openedx/core/djangoapps/contentserver/views.py

```diff
@@ -9,50 +9,357 @@
-around the latter.
-
-A waffle flag chooses whether to allow the middleware to handle the request, or
-whether to pass the request along to the view. Why? Because we might be relying
-by accident on some weird behavior inherent to misusing a middleware this way,
-and we need a way to quickly switch back if we encounter problems.
-
-If the view works, we can move all of StaticContentServerImpl directly into the
-view and drop the middleware and the waffle flag.
+We've turned it into a proper view, with a few warts remaining:
+
+- The view implementation is all bundled into a StaticContentServer class that
+  doesn't appear to have any state. The methods could likely just be extracted
+  as top-level functions.
+- All three urlpatterns are registered to the same view, which then has to
+  re-parse the URL to determine which pattern is in effect. We should probably
+  have 3 views as entry points.
 """
-from django.http import HttpResponseNotFound
+import datetime
+import logging
+
+from django.http import (
+    HttpResponse,
+    HttpResponseBadRequest,
+    HttpResponseForbidden,
+    HttpResponseNotFound,
+    HttpResponseNotModified,
+    HttpResponsePermanentRedirect
+)
 from django.views.decorators.http import require_safe
 from edx_django_utils.monitoring import set_custom_attribute
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.locator import AssetLocator
+
+from common.djangoapps.student.models import CourseEnrollment
+from openedx.core.djangoapps.header_control import force_header_for_response
+from xmodule.assetstore.assetmgr import AssetManager
+from xmodule.contentstore.content import XASSET_LOCATION_TAG, StaticContent
+from xmodule.exceptions import NotFoundError
+from xmodule.modulestore import InvalidLocationError
+from xmodule.modulestore.exceptions import ItemNotFoundError
 
-from .middleware import CONTENT_SERVER_USE_VIEW, IMPL
+from .caching import get_cached_content, set_cached_content
+from .models import CdnUserAgentsConfig, CourseAssetCacheTtlConfig
 
 
 @require_safe
 def course_assets_view(request):
     """
     Serve course assets to end users. Colloquially referred to as "contentserver."
     """
-    set_custom_attribute('content_server.handled_by.view', True)
-
-    if not CONTENT_SERVER_USE_VIEW.is_enabled():
-        # Should never happen; keep track of occurrences.
-        set_custom_attribute('content_server.view.called_when_disabled', True)
-        # But handle the request anyhow.
-
-    # We'll delegate request handling to an instance of the middleware
-    # until we can verify that the behavior is identical when requests
-    # come all the way through to the view.
-    response = IMPL.process_request(request)
-
-    if response is None:
-        # Shouldn't happen
-        set_custom_attribute('content_server.view.no_response_from_impl', True)
-        return HttpResponseNotFound()
-    else:
-        return response
+    return IMPL.process_request(request)
```

Comment:
Nit: I hate the name `IMPL`. I prefer something like `STATIC_CONTENT_SERVER_INSTANCE`, or better yet, just replace it with `StaticContentServer()`, assuming that instantiation is negligible.

Reply:
Yeah, no argument there! But I think the next step is to actually just get rid of `StaticContentServer` entirely, as it's a class with no state and all the methods are just helper functions that could be dumped into the module, top-level.

Reply:
Yes. The only difference is that the former could have been done in 1 minute, and the latter may never actually happen (unless you are planning more clean-up). Oh well. :)

Reply:
Technically, it could have been done in 25 minutes, because of the length of time it takes to re-run tests on edx-platform. :-P Sorry, I just really didn't want to do another cycle of that. I do have more cleanup, planned, though -- this was intended as a minimum possible refactor to get rid of the middleware.